pyccl/tests/test_covs.py

```diff
@@ -41,59 +42,126 @@ def get_tracer():
 
     # Different tracers
-    cov = ccl.angular_cl_cov_cNG(COSMO, tr, tr, ls, tsp,
-                                 cltracer3=tr, cltracer4=tr)
+    cov = cov_f(ls, cltracer3=tr, cltracer4=tr)
     assert np.all(np.fabs(cov/cov_p-1).flatten() < 1E-5)
 
     # Different ells
-    cov = np.array([ccl.angular_cl_cov_cNG(COSMO, tr, tr, ls, tsp,
-                                           ell2=l)
+    cov = np.array([cov_f(ls, ell2=l)
                     for l in ls])
     assert np.all(np.fabs(cov/cov_p-1).flatten() < 1E-5)
 
     # Different ells (transpose)
-    cov = np.array([ccl.angular_cl_cov_cNG(COSMO, tr, tr, l, tsp,
-                                           ell2=ls)
+    cov = np.array([cov_f(l, ell2=ls)
                     for l in ls]).T
     assert np.all(np.fabs(cov/cov_p-1).flatten() < 1E-5)
 
     # All scalars
-    cov = np.array([[ccl.angular_cl_cov_cNG(COSMO, tr, tr, l1, tsp,
-                                            ell2=l2)
+    cov = np.array([[cov_f(l1, ell2=l2)
                      for l1 in ls]
                     for l2 in ls])
     assert np.all(np.fabs(cov/cov_p-1).flatten() < 1E-5)
 
 
-def test_cov_cNG_errors():
+@pytest.mark.parametrize("typ", ['cNG', 'SSC'])
+def test_cov_NG_errors(typ):
+    if typ == 'cNG':
+        cov_f = ccl.angular_cl_cov_cNG
+    elif typ == 'SSC':
+        cov_f = ccl.angular_cl_cov_SSC
+
     tsp = get_tk3d(1, 1)
     tr = get_tracer()
     ls = np.array([2., 20., 200.])
 
-    assert_raises(ValueError, ccl.angular_cl_cov_cNG,
+    assert_raises(ValueError, cov_f,
                   COSMO, tr, tr, ls, tsp,
                   integration_method='cag_cuad')
 
-    assert_raises(ValueError, ccl.angular_cl_cov_cNG,
+    assert_raises(ValueError, cov_f,
                   COSMO, tr, tr, ls, tr)
+
+
+def test_Sigma2B():
+    # Check projected variance calculation against
+    # explicit integration
+    from scipy.special import jv
+    from scipy.integrate import simps
+
+    fsky = 0.1
+    # Default sampling
+    a, s2b_a = ccl.sigma2_B_disc(COSMO, fsky=fsky)
+    idx = (a > 0.5) & (a < 1)
+
+    a_use = a[idx]
+    # Input sampling
+    s2b_b = ccl.sigma2_B_disc(COSMO, a=a_use, fsky=fsky)
+    # Scalar input sampling
+    s2b_c = np.array([ccl.sigma2_B_disc(COSMO, a=a, fsky=fsky)
+                      for a in a_use])
+
+    # Alternative calculation
+    chis = ccl.comoving_radial_distance(COSMO, a_use)
+    Rs = np.arccos(1-2*fsky)*chis
+
+    def integrand(lk, a, R):
+        k = np.exp(lk)
+        x = k*R
+        w = 2*jv(1, x)/x
+        pk = ccl.linear_matter_power(COSMO, k, a)
+        return w*w*k*k*pk/(2*np.pi)
+
+    lk_arr = np.log(np.geomspace(1E-4, 1E1, 1024))
+    s2b_d = np.array([simps(integrand(lk_arr, a, R), x=lk_arr)
+                      for a, R in zip(a_use, Rs)])
+
+    assert np.all(np.fabs(s2b_b/s2b_a[idx]-1) < 1E-10)
+    assert np.all(np.fabs(s2b_c/s2b_a[idx]-1) < 1E-10)
+    assert np.all(np.fabs(s2b_d/s2b_a[idx]-1) < 1E-3)
```

Comment:
This is as good as it gets with a quick Simpsons integration on `scipy`. We could probably improve it by slicing finer in `k`, but 1E-3 should be good enough for these purposes (since the SSC itself is already a small contribution to the total covariance)